mob/src/main/java/com/github/phantazmnetwork/mob/spawner/BasicMobSpawner.java

```diff
@@ -3,64 +3,169 @@
+        ElementContext context = contextManager.makeContext(model.getNode());
+        addGoals(context, dependencyProvider, neuralEntity, model);
+        Map<Key, Collection<Skill>> triggers = createTriggers(context, dependencyProvider, neuralEntity, model);
 
+        PhantazmMob mob = new PhantazmMob(model, neuralEntity, triggers);
         mobStore.registerMob(mob);
         return mob;
     }
 
-    private void setDisplayName(@NotNull NeuralEntity neuralEntity, @NotNull MobModel model) {
-        model.getDisplayName().ifPresent(displayName -> {
-            neuralEntity.setCustomName(displayName);
-            neuralEntity.setCustomNameVisible(true);
-        });
+    private void setEntityMeta(@NotNull NeuralEntity neuralEntity, @NotNull MobModel model) {
+        EntityMeta meta = neuralEntity.getEntityMeta();
+        ConfigNode metaNode = model.getMetaNode();
+        for (Method method : meta.getClass().getMethods()) {
+            if (!Modifier.isPublic(method.getModifiers()) || method.getReturnType() != void.class) {
+                continue;
+            }
+
+            Parameter[] parameters = method.getParameters();
+            if (parameters.length != 1) {
+                continue;
+            }
+
+            String methodName = method.getName();
+            if (!methodName.startsWith("set")) {
+                continue;
+            }
+            String key;
+            if (methodName.length() > 3) {
+                key = Character.toLowerCase(methodName.charAt(3)) + methodName.substring(4);
+            }
+            else {
+                key = methodName.substring(3);
+            }
+
+            ConfigElement element = metaNode.getElementOrDefault((ConfigElement)null, key);
+            if (element == null) {
+                continue;
+            }
+
+            Parameter parameter = parameters[0];
+            NotNull[] notNulls = parameter.getAnnotationsByType(NotNull.class);
+            Nullable[] nullables = parameter.getAnnotationsByType(Nullable.class);
+            Class<?> type = parameter.getType();
+            boolean optional = !type.isPrimitive() && nullables.length != 0 || notNulls.length < 1;
+
+            ConfigProcessor<?> processor = processorMap.get(BooleanObjectPair.of(optional, type.getName()));
+            if (processor == null) {
+                continue;
+            }
+
+            Object data;
+            try {
+                data = processor.dataFromElement(element);
+            }
+            catch (ConfigProcessException e) {
+                e.printStackTrace();
+                continue;
+            }
+            if (data instanceof Optional<?> dataOptional) {
+                data = dataOptional.orElse(null);
+            }
+            try {
+                method.invoke(meta, data);
+            }
+            catch (IllegalAccessException | InvocationTargetException e) {
+                e.printStackTrace();
```

Comment:
another potential case for logging, alternatively, rethrowing the exception, don't really see a reason to swallow it here?